design/authenticators/authn-k8s/inject_client_cert_solution_design.md

```diff
@@ -0,0 +1,195 @@
+# Solution Design - Inject Client Cert fix
+[//]: # "Change the title above from "Template" to your design's title"
+
+## Table of Contents
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+## Useful links
+[//]: # "Add links that may be useful for the reader"
+
+- [Existing PR with solution](https://github.com/cyberark/conjur/pull/1789)
+  - In this solution we wait for the timeout to get an answer and if we didn't
+    get one we create another WebSocket with an "ls" command to verify that the
+    cert is injected.
+- [Known issue in k8s exec](https://github.com/kubernetes/kubernetes/issues/89899)
+
+## Background
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+We have an issue with the "inject client cert" process where we don't always return a correct response to the client regarding the injection success. The injection is done via a WebSocket to Kubectl Exec where we use a "tar" command and the cert content in STDIN. There is [an issue](https://github.com/kubernetes/kubernetes/issues/89899) in k8s where messages that are sent via WebSockets to Kubectl Exec are left hanging without returning a response when using STDIN. 
```

Comment:
When you say kubectl, you actually mean the Kubernetes API server or the Ruby kubeclient?

Reply:
the former. i updated the doc to say `via a WebSocket to the Kubernetes API server`